### PR TITLE
Added run_single method to run single migration by version and direction

### PR DESCRIPTION
**Pull request by opya**
The new method was added only for TimestampMigrator because I'm not sure that this is the correct way to do it.

Reference: https://github.com/jeremyevans/sequel/discussions/2045

### CODE REVIEW

**Review by jeremyevans — CHANGES_REQUESTED** [review]
I think the API should be different:

```ruby
  Sequel::TimestampMigrator.run_single(DB, '/path/to/filename', opts)
```

With this definition:

```ruby
  def TimestampMigrator.run_single(db, path, opts=OPTS)
    new(db, File.dirname(path), opts).run_single(path, opts[:direction])
  end
```

This avoids a lot of the unnecessary code in this pull request.  It also handles the case of multiple migrations with the same version (allowed by the TimestampMigrator). Can you switch to that approach?

**Comment by opya** [comment]
sorry for the multiple commits, i think that the requested comments were addressed, sorry if forgot something

File: lib/sequel/extensions/migration.rb [hunk 1]
```diff
@@ -585,6 +589,10 @@ def run
       target
     end
 
+    def run_single(path, direction)
```
Comment (jeremyevans) [inline lib/sequel/extensions/migration.rb:592]: This code should be removed.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Looking better.  Please see some additional requested changes.

**Review by jeremyevans — COMMENTED** [review]
This looks good.  I'll try to merge and test on Monday.

**Comment by opya** [comment]
ok, thank you

**Comment by jeremyevans** [comment]
Squash merged at 422f16040a09e5bf5f6298121f5e246f91418ea4

**Pull request closed without merging**